includes/rest/class-server.php

```diff
@@ -170,4 +172,48 @@ public static function request_parameter_order( $order, $request ) {
 			'defaults',
 		);
 	}
+
+	/**
+	 * Filters the REST API response to properly handle the ActivityPub error formatting.
+	 *
+	 * @see https://codeberg.org/fediverse/fep/src/branch/main/fep/c180/fep-c180.md
+	 *
+	 * @param WP_HTTP_Response $response Result to send to the client. Usually a `WP_REST_Response`.
+	 * @param WP_REST_Server   $server   Server instance.
+	 * @param WP_REST_Request  $request  Request used to generate the response.
+	 *
+	 * @return WP_HTTP_Response The filtered response.
+	 */
+	public static function filter_output( $response, $server, $request ) {
+		$route = $request->get_route();
+
+		// Check if it is an activitypub request and exclude webfinger and nodeinfo endpoints.
+		if ( ! \str_starts_with( $route, '/' . ACTIVITYPUB_REST_NAMESPACE ) ) {
+			return $response;
+		}
+
+		// Only alter responses that return an error status code.
+		if ( $response->get_status() < 300 ) {
```

Comment:
```suggestion
		if ( $response->get_status() < 400 ) {
```
I think changing that would be good, before merging